### PR TITLE
reduce multiple identifies

### PR DESCRIPTION
**Pull request by chrisnojima**

### CODE REVIEW

File: shared/actions/tracker.js [hunk 1]
```diff
@@ -79,6 +79,7 @@ export function getProfile (username: string): TrackerActionCreator {
   return (dispatch, getState) => {
     // If we have a pending identify no point in firing off another one
     if (getState().tracker.pendingIdentifies[username]) {
+      console.log('Bailing on simultaneous getProfile', username)
```
Comment (chrisnojima) [inline shared/actions/tracker.js]: just so we can understand if this is causing issues

File: shared/actions/tracker.js [hunk 2]
```diff
@@ -130,19 +131,13 @@ export function triggerIdentify (uid: string = '', userAssertion: string = ''
   }
 
   return (dispatch, getState) => new Promise((resolve, reject) => {
-    dispatch(pendingIdentify(userAssertion || uid, true))
```
Comment (chrisnojima) [inline shared/actions/tracker.js]: this is handled in the callback map now

File: shared/actions/tracker.js [hunk 3]
```diff
@@ -425,16 +419,43 @@ function updatePGPKey (username: string, pgpFingerprint: Buffer, kid: string): A
 const sessionIDToUsername: { [key: number]: string } = {}
 // TODO: if we get multiple tracker calls we should cancel one of the sessionIDs, now they'll clash
 function serverCallMap (dispatch: Dispatch, getState: Function, skipPopups: boolean = false, onFinish: ?() => void): incomingCallMapType {
+  // if true we already have a pending call so lets skip a ton of work
+  let username
+  let clearPendingTimeout
+  let alreadyPending = false
+
+  const requestIdle = f => {
```
Comment (chrisnojima) [inline shared/actions/tracker.js]: just a wrapper so we check if its pending and only do 'our' work if its not a duplicate (we always respond to the server)

File: shared/actions/tracker.js [hunk 4]
```diff
@@ -425,16 +419,43 @@ function updatePGPKey (username: string, pgpFingerprint: Buffer, kid: string): A
 const sessionIDToUsername: { [key: number]: string } = {}
 // TODO: if we get multiple tracker calls we should cancel one of the sessionIDs, now they'll clash
 function serverCallMap (dispatch: Dispatch, getState: Function, skipPopups: boolean = false, onFinish: ?() => void): incomingCallMapType {
+  // if true we already have a pending call so lets skip a ton of work
+  let username
```
Comment (chrisnojima) [inline shared/actions/tracker.js]: store the username locally instead of looking it up with session all the time

File: shared/actions/tracker.js [hunk 5]
```diff
@@ -626,8 +637,19 @@ function serverCallMap (dispatch: Dispatch, getState: Function, skipPopups: bool
           },
         })
 
+        // Doing a non-tracker so explicitly cleanup instead of using the timeout
+        if (skipPopups) {
```
Comment (chrisnojima) [inline shared/actions/tracker.js]: if its an incoming call (tracker) we actually let the timeout happen so we have some time where we ignore incoming requests.

File: shared/common-adapters/icon.desktop.js [hunk 6]
```diff
@@ -1,13 +1,13 @@
...
-import React, {Component} from 'react'
+import React, {PureComponent} from 'react'
 import type {$Exact} from '../constants/types/more'
 import type {Props} from './icon'
 import {FontIcon} from 'material-ui'
 import {globalStyles, globalColors} from '../styles'
 import {resolveImageAsURL} from '../../desktop/resolve-root'
 
-class Icon extends Component<void, $Exact<Props>, void> {
+class Icon extends PureComponent<void, $Exact<Props>, void> {
```
Comment (chrisnojima) [inline shared/common-adapters/icon.desktop.js]: try and speed this up a little

File: shared/common-adapters/user-proofs.desktop.js [hunk 7]
```diff
@@ -51,6 +53,9 @@ class ProofRow extends Component<void, ProofRowProps, ProofRowState> {
       hovering: false,
       popupMenuPosition: {},
     }
+
+    this._onMouseEnter = () => this.setState({hovering: true})
```
Comment (chrisnojima) [inline shared/common-adapters/user-proofs.desktop.js]: instead of making the closures on the render call we can just store it on ourselves

**Comment by chrisnojima** [comment]
@keybase/react-hackers take a peek

**Comment by chrisnojima** [comment]
@keybase/react-hackers bump

**Comment by chromakode** [comment]
Would you please merge master so the changes from https://github.com/keybase/client/pull/4207 are not in the diff?

**Comment by MarcoPolo** [comment]
👍

**Pull request merged**